framework/src/userobject/SolutionUserObject.C

```diff
@@ -271,24 +271,34 @@ SolutionUserObject::readExodusII()
       if (std::find(all_elemental.begin(), all_elemental.end(), var_name) != all_elemental.end())
         _elemental_variables.push_back(var_name);
       if (std::find(all_scalar.begin(), all_scalar.end(), var_name) != all_scalar.end())
-        _scalar_variables.push_back(var_name);
+        // Check if the scalar matches any field variables, and ignore the var if it does. This
+        // means its a Postprocessor.
+        if (std::find(begin(_nodal_variables), end(_nodal_variables), var_name) ==
+            _nodal_variables.end())
+          _scalar_variables.push_back(var_name);
     }
   }
   else
   {
     _nodal_variables = all_nodal;
     _elemental_variables = all_elemental;
-    _scalar_variables = all_scalar;
+
+    for (std::string var_name : all_scalar)
+      // Check if the scalar matches any field variables, and ignore the var if it does. This means
+      // its a Postprocessor.
+      if (std::find(begin(_nodal_variables), end(_nodal_variables), var_name) ==
+          _nodal_variables.end())
+        _scalar_variables.push_back(var_name);
   }
 
   // Add the variables to the system
-  for (const auto & var_name : _nodal_variables)
+  for (const std::string & var_name : _nodal_variables)
     _system->add_variable(var_name, FIRST);
 
-  for (const auto & var_name : _elemental_variables)
+  for (const std::string & var_name : _elemental_variables)
     _system->add_variable(var_name, CONSTANT, MONOMIAL);
 
-  for (const auto & var_name : _scalar_variables)
+  for (const std::string & var_name : _scalar_variables)
```

Comment:
why the change?

Reply:
i mean, auto is slower than specifying it directly, no?

Reply:
not really. It s decided at compile time, and if we ever change the type we wont need to change all the code if it s auto, but we will need to if it s std::string